pandas/core/indexes/numeric.py

```diff
@@ -107,7 +107,7 @@ def _can_hold_na(self) -> bool:
         else:
             return False
 
-    @cache_readonly
+    @property
```

Comment:
`_engine_type` is only called once per instance, so cache_readonly just complicated the cache